.circleci/config.yml

```diff
@@ -747,6 +747,46 @@ jobs:
             make -C ${MAGMA_ROOT}/cwf/gateway/integ_tests precommit
       - magma_slack_notify
 
+  cwag-build-check:
+    machine:
+      image: ubuntu-2004:202010-01
+      docker_layer_caching: true
+    resource_class: large
+    environment:
+      MAGMA_ROOT: /home/circleci/project
+    steps:
+      - checkout
+      - build/determinator:
+          <<: *cwf_build_verify
+      - run:
+          name: Build CWAG containers
+          command: |
+            cd $MAGMA_ROOT/cwf/gateway/docker
+            docker-compose build --parallel
+      - run:
+          name: Generate test certs
+          command: |
+            sudo apt install openssl
+            mkdir -p $MAGMA_ROOT/cwf/gateway/docker/.test_certs
+            cd $MAGMA_ROOT/cwf/gateway/docker/.test_certs
+            touch snowflake
+            openssl genrsa -out rootCA.key 2048
+            openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 365000 \
+              -out rootCA.pem -subj "/C=US/CN=rootca.magma.test"
+      - run:
+          name: Bring up CWAG containers
+          command: |
+            cd $MAGMA_ROOT/cwf/gateway/docker
+            docker-compose --env-file=./.ci_env up -d
+      - run:
+          name: Check health (except pipelined and redirectd)
+          command: |
+            docker ps --format "{{.Names}}	{{.Status}}" \
+              | grep Restarting \
+              | grep --invert-match 'pipelined\|redirectd'
+            EXITCODE=$?
+            $EXITCODE -eq 1 # grep returns 1 when finding nothing
+
```

Comment:
Note you need to the service hold for some seconds. IF it is crashlooping you will see that `restarting` just shows up for very short time. 

For my PR on FEG I created a script that checks for it's health for few seconds
https://github.com/magma/magma/pull/6288/files

I am just waiting for approval